prediction_market_agent/deploy/deploy.py

```diff
@@ -0,0 +1,137 @@
+from cron_validator import CronValidator
+from enum import Enum
+import os
+from pydantic import BaseModel
+import requests
+import shutil
+import subprocess
+import tempfile
+import time
+
+from prediction_market_agent.data_models.market_data_models import AgentMarket
+from prediction_market_agent.deploy.utils import (
+    export_requirements_from_toml,
+    gcloud_delete_function_cmd,
+    gcloud_deploy_cmd,
+    gcloud_schedule_cmd,
+    get_gcloud_function_uri,
+    get_gcloud_id_token,
+)
+from prediction_market_agent.markets.all_markets import (
+    MarketType,
+    get_binary_markets,
+    place_bet,
+)
+from prediction_market_agent.tools.betting_strategies import get_tiny_bet
+from prediction_market_agent.utils import APIKeys
+
+
+class DeploymentType(str, Enum):
+    GOOGLE_CLOUD = "google_cloud"
+    LOCAL = "local"
+
+
+class DeployableAgent(BaseModel):
+    def pick_markets(self, markets: list[AgentMarket]) -> list[AgentMarket]:
+        if len(markets) > 1:
+            return markets[:1]
+        return markets
+
+    def answer_binary_market(self, market: AgentMarket) -> bool:
+        raise NotImplementedError("This method should be implemented by the subclass")
+
+    def deploy(
```

Comment:
Is this method used somewhere? 🤔

Reply:
I guess it's a placeholder for now until we have a nice way of merging